### DISCUSSION TITLE
Question on coupling between phase field and tensor mechanics

### DISCUSSION THREAD

**Discussion by JingShuShi** [idaholab/moose#17919] [Q&A Modules: Phase field] [answered]
Dear MOOSE team,

I have few general questions regarding the coupling between phase field and tensor mechanics:

By default, a single phase field simulation uses Eulerian/spatial meshes that don't deform, while a single tensor mechanics simulation uses Lagrangian/material meshes that will deform. When they are coupled together, which mesh type is used?

If the material meshes are used, how do we think about the mass conservation in a coupled RVE? For instance, if we have solid and gas phases in a RVE, the solid will flow in/out of the domain due to the mass diffusion, thus the total mass can not be conserved inside the RVE, which gives problem justifying the mass periodicity at the boundary. 
If spatial meshes are used, how do the meshes deform? Or how do phase field variables influence the traction at the boundary?

Hope you could give me some insights/hints to the questions. Thanks in advance.

Best regards,
Hao

**Comment by jiangwen84** [comment, marked as answer]
Can you formulate your phase field equations on a deformed configuration? You can potentially solve phase field equations on a deformed mesh by setting `use_displaced_mesh = true`.

**Reply by hugary1995** [reply]
I think it depends on the phase field model. If his phase field describes an Eulerian interface, then he has to use some Eulerian-Lagrangian methods to couple with tensor_mechanics. If his phase field is Lagrangian, there is no issue.

I doubt this is related to `use_displaced_mesh`.

**Reply by JingShuShi** [reply]
@hugary1995 I am thinking of solid diffusion in a polycrystalline structure/RVE, where grains are diffusing by concentration field and moving as a rigid body at the same time. The diffusional interface is by default a Lagrangian problem, but to formulate and track this interface this would be very cumbersome and time consuming. Whereas phase field approach bring this problem into Eulerian description and avoid tracking the interface specifically. Please correct me if my thoughts are off somewhere...

@jiangwen84 I noticed  ```use_displaced_mesh``` can be used in tensor mechanics, but not sure how this will affect the coupling.
My key problem is in thinking through how the phase field simulation can change the RVE volume via diffusion and advection, i.e., boundaries deforms according to phase field variables, such that I don't need to think of remeshing every timestep to find where the boundary sits, then we could evaluate properties with the right RVE volume. Then I got stuck in this Eulerian-Lagrangian RVE setting...

**Reply by SudiptaBiswas** [reply]
You shouldn't need `use_displaced_mesh = true` unless you have large deformation. For small deformations, we have coupled phase-field with mechanics without any issues. 
Which phase-field model are you using?  In general, phase-field does not change the volume and you shouldn't need to track the interface. Maybe explain what you are trying to do in a bit more detail.

**Reply by JingShuShi** [reply]
Okay, I want to simulate the densification in a solid state free sintering with a RVE, then we could apply mechanical loads with different micro-structure at different densities to check homogenized mechanical properties. If I consider free sintering without external load in a periodic RVE, there are in general two ways to achieve densification: 1. fix domain and let mass flow into this domain at the boundary; 2. keep total solid mass constant and shrink the total domain volume. As material will flow/move inside the domain due to diffusion and advection, the domain boundary should also move accordingly. This is where I got into Eulerian-Lagrangian settings of the RVE. But you pointed out in general the phase-field does not change the volume, would it still be possible to achieve densification using the phase-field method?

**Reply by SudiptaBiswas** [reply]
The densification from a phase-field sintering simulation is calculated in terms of the reduction in free surfaces and/or internal porosity. You can find some of our sintering work in the following papers,

https://www.sciencedirect.com/science/article/pii/S2352431616300438

https://link.springer.com/article/10.1007/s10853-017-1846-3

https://www.sciencedirect.com/science/article/pii/S0927025619305877

**Reply by JingShuShi** [reply]
Thanks for sharing the interesting papers, they are really helpful. In the first two papers, the mechanical loads are applied to the boundary with prescribed displacements instead of the traction boundary with fixed loads, the solid and void phases are lumped together like a composite material and the composite elasticity tensor is used in the mechanical equilibrium calculation. If one uses the stress at the boundary instead, would it lead to some convergence problem like mesh distortion as void phase has very low elasticity so they could not carry any loads? In your second paper, the last figure with 2 particles deforming in 3D looks very nice, but they deform so much at the end and would small deformation assumption still be valid there?
 
The last paper gives a good overview on the density evaluation methods during sintering, the MC algorithm to evaluate the relative density seems to be a good approach. But all the methods need a post-process step and seem difficult to correlate to homogenization. If one wants to shrink the domain with a global strain that corresponds to local particle movements, how would you suggest to achieve this in MOOSE?

**Reply by SudiptaBiswas** [reply]
Yes, instead of the applied displacement, you can use the pressure boundary condition. To avoid numerical issues, you will have to assign a small non-zero stiffness for the void.  Yes, in the 3D example, the deformation was kept within the elastic limit.

You can change the volume of an RVE using the `GlobalStrain` approach. See here, https://www.sciencedirect.com/science/article/pii/S0168874X20301165

**Reply by JingShuShi** [reply]
Yes, I get you applied small non-zero stiffness for the void phase, the only thing I am still puzzling is that when you apply a pressure load on the top of a 2D domain with particles in the center and not touching the boundary, the void phase will take the load first and deform, then transfer to solid phase. If the void stiffness is very small, the deformation on void elements will be much larger than on solid phase, or in other words, void phase could not take any loads, only solid phase can, how is the problem overcome in this setting? Maybe I misunderstood how the two phase composite works in mechanical module...

The second paper looks very promising, is there a way to correlate particle rigid body motion to the ```GlobalStrain```? Because in all the paper it clearly stated that rigid body motion is the key point to the volume shrinkage during sintering... If this feature is not there, maybe I could contribute in developing it, then I probably will need more help/guidance from the expert like you.

**Reply by SudiptaBiswas** [reply]
The stiffness of the void was selected in a way to avoid such issues. Also, for small displacement problems, the calculation is done on the undisplaced mesh. No, `GlobalStrain` has not been connected to rigid-body motion.

**Reply by JingShuShi** [reply]
Thanks for explanations, maybe connecting rigid-body motion with GlobalStrain is a way out to my problem, it will be a GlobalStrain only coming from stress free rigid-body motion, or purely due to diffusion induced rigid-body motion, how would you suggest to do this in MOOSE?
